### ISSUE TITLE
Types broken in 3.0.0

### ISSUE THREAD

**Issue by qtiki** [pixijs/pixi-sound#107]
Type definitions are broken in 3.0.0:
```
node_modules/pixi-sound/pixi-sound.d.ts:263:13 - error TS2416: Property 'init' in type 'HTMLAudioInstance' is not assignable to the same property in base type 'IMediaInstance'.
  Type '(media: HTMLAudioMedia) => void' is not assignable to type '(parent: IMedia) => void'.
    Types of parameters 'media' and 'parent' are incompatible.
      Type 'IMedia' is missing the following properties from type 'HTMLAudioMedia': parent, source, eventNames, listeners, and 8 more.

263             init(media: HTMLAudioMedia): void;
                ~~~~

node_modules/pixi-sound/pixi-sound.d.ts:292:13 - error TS2416: Property 'create' in type 'HTMLAudioMedia' is not assignable to the same property in base type 'IMedia'.
  Type '() => HTMLAudioInstance' is not assignable to type '() => IMediaInstance'.
    Property 'set' is missing in type 'HTMLAudioInstance' but required in type 'IMediaInstance'.

292             create(): HTMLAudioInstance;
                ~~~~~~

  node_modules/pixi-sound/pixi-sound.d.ts:159:9
    159         set(name:'speed'|'volume'|'muted'|'loop'|'paused', value:number|boolean): this;
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    'set' is declared here.

node_modules/pixi-sound/pixi-sound.d.ts:339:13 - error TS2416: Property 'init' in type 'WebAudioInstance' is not assignable to the same property in base type 'IMediaInstance'.
  Type '(media: WebAudioMedia) => void' is not assignable to type '(parent: IMedia) => void'.
    Types of parameters 'media' and 'parent' are incompatible.
      Type 'IMedia' is missing the following properties from type 'WebAudioMedia': parent, source, buffer, nodes

339             init(media: WebAudioMedia): void;
                ~~~~

node_modules/pixi-sound/pixi-sound.d.ts:346:13 - error TS2416: Property 'create' in type 'WebAudioMedia' is not assignable to the same property in base type 'IMedia'.
  Type '() => WebAudioInstance' is not assignable to type '() => IMediaInstance'.
    Property 'set' is missing in type 'WebAudioInstance' but required in type 'IMediaInstance'.

346             create(): WebAudioInstance;
                ~~~~~~

  node_modules/pixi-sound/pixi-sound.d.ts:159:9
    159         set(name:'speed'|'volume'|'muted'|'loop'|'paused', value:number|boolean): this;
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    'set' is declared here.
```

I think this is because the types don't include any of the original private members so TypeScript doesn't know that the types are in fact compatible.

I tried looking at how the types are generated in the first place but couldn't find any automation. Have you considered using tsconfig `declarations` compiler option to generate them automatically? I'm not sure if that can be made to "play nice" with rollup though.

**Comment by bigtimebuddy** [comment]
I can reproduce this, will work on a fix ASAP.

**Comment by bigtimebuddy** [comment]
Published v3.0.1
